src/TeamSyncer.ts

```diff
@@ -368,24 +383,53 @@ export class TeamSyncer {
             return;
         }
 
+        await this.notifyAdmins(`${teamId} removed channel ${channelId}, bridged room: ${room.MatrixRoomId}`);
+        await this.shutDownBridgedRoom("deleted", room.MatrixRoomId);
+    }
+
+    public async onChannelArchived(teamId: string, channelId: string): Promise<void> {
+        log.info(`${teamId} archived channel ${channelId}`);
+        if (!this.getTeamSyncConfig(teamId, "channel", channelId)) {
+            return;
+        }
+        const room = this.main.rooms.getBySlackChannelId(channelId);
+        if (!room) {
+            log.warn("Not unlinking channel, no room found");
+            return;
+        }
+
+        await this.notifyAdmins(`${teamId} archived channel ${channelId}, bridged room: ${room.MatrixRoomId}`);
+        await this.shutDownBridgedRoom("archived", room.MatrixRoomId);
+    }
+
+    public async notifyAdmins(message: string) {
+        if (this.adminRoom) {
+            await this.main.botIntent.sendMessage(this.adminRoom, {
+                msgtype: "m.notice",
+                body: message,
+            });
+        }
+    }
+
+    public async shutDownBridgedRoom(reason: string, roomId: string) {
         try {
-            await this.main.botIntent.sendMessage(room.MatrixRoomId, {
+            await this.main.botIntent.sendMessage(roomId, {
                 msgtype: "m.notice",
-                body: "The Slack channel bridged to this room has been deleted.",
+                body: `The Slack channel bridged to this room has been \`${reason}\`.`,
```

Comment:
Could use `'` to avoid escaping.

```suggestion
                body: `The Slack channel bridged to this room has been '${reason}'.`,
```